ext/gmp/gmp.c

```diff
@@ -1350,22 +1350,20 @@ ZEND_FUNCTION(gmp_pow)
 		RETURN_THROWS();
 	}
 
-    double powmax = log((double)ZEND_LONG_MAX);
+	double max_bits = 16000; // TODO: value is very small, but passed current test suite
 
 	if (Z_TYPE_P(base_arg) == IS_LONG && Z_LVAL_P(base_arg) >= 0) {
 		INIT_GMP_RETVAL(gmpnum_result);
-		if ((log(Z_LVAL_P(base_arg)) * exp) > powmax) {
+		if ((log2(Z_LVAL_P(base_arg)) * exp) > max_bits) {
```

Comment:
For parity with the else branch, this should be
```suggestion
		if ((floor(log2(Z_LVAL_P(base_arg))) * exp) > max_bits) {
```

But that would fail the last check in gmp_pow_fpe.phpt.